btf/btf.go

```diff
@@ -655,6 +655,31 @@ func (s *Spec) TypeByName(name string, typ interface{}) error {
 	return nil
 }
 
+// TypesIterator iterates over types of a given spec.
+type TypesIterator struct {
+	spec *Spec
+	// The index of the last visited type in the spec.
+	Index int
```

Comment:
Do you need to expose the index? I'd prefer to keep it an implementation detail for now.